kong/plugins/tcp-log/handler.lua

```diff
@@ -52,6 +52,28 @@ local TcpLogHandler = {
 }
 
 
+function TcpLogHandler:access(conf)
+  if conf.log_body then
+    local body, err = kong.request.get_raw_body()
+    if err then
+      kong.log.err(err)
+      kong.ctx.plugin.request_body = ""
+    else
+      kong.ctx.plugin.request_body = body
+    end
+    kong.ctx.plugin.response_body = ""
```

Comment:
```suggestion
    kong.ctx.plugin.response_body = {}
```

Reply:
I made changes, but some tests failed:
![image](https://user-images.githubusercontent.com/13045944/96568342-6b860000-12fa-11eb-9734-88db6ce67325.png)

Corresponding code:
![image](https://user-images.githubusercontent.com/13045944/96570521-10a1d800-12fd-11eb-89f5-09c703299a4a.png)

This function is used to check whether the temporary log path exists and get its size, but it times out after waiting 10 seconds. Is this accidental?

Reply:
shouldn't be related

Reply:
I have made changes as proposed. Can this pr now be merged, or does it require all tests to pass?

Reply:
looked into the failures, and they seem to be related; https://travis-ci.org/github/Kong/kong/jobs/737346018#L4696-L4707

They are all logging plugins that fail, and some changes you made are shared by all those plugins. Maybe try rebase on `next` first, and try and run those tests again.

Reply:
Thank you for suggestions. Before I push again, I have rebase `next`, and currently my branch contains all the commits of the `next` branch and two commits ahead:

![image](https://user-images.githubusercontent.com/13045944/96696159-0810d680-13bd-11eb-8ba5-112643c7a583.png)

Observe the failed test, the [01-tcp-log](https://travis-ci.org/github/Kong/kong/jobs/737346018#L477) failed test is the original test, not my newly added test case, and the log body switch is not enabled in this test case. The error line 219 is `local ok, res = thread:join()` and the `ok` is false, which is related to the previous `thread:join()` timeout: https://travis-ci.org/github/Kong/kong/jobs/737346018#L471. 
![image](https://user-images.githubusercontent.com/13045944/96697918-0ea04d80-13bf-11eb-8e03-c13e2bb99b08.png)


The failure case of [02-udp-log](https://travis-ci.org/github/Kong/kong/jobs/737346018#L542) and [07-loggly](https://travis-ci.org/github/Kong/kong/jobs/737346018#L645) are also related to the result of `local ok, res = thread:join()`. The [03-http-log](https://travis-ci.org/github/Kong/kong/jobs/737346018#L542) and [04-file-log](https://travis-ci.org/github/Kong/kong/jobs/737346018#L571) failed test related to `wait_until() timeout (after delay 10s)` etc.

I don't see a rebuild button to re-run the test, and there is no new submission to trigger the test. Could you help me run the test again?

Reply:
if you look at the test code (not in the helpers.lua) you probably see that it fires a request, and then wait for a specific message to appear in the logs. My guess is that the message format changed, and hence the timeout occurs while waiting. So the test most likely must be adjusted for the new output, or if the output doesn't match because it is erroneous, the code must be adapted obviously.

Reply:
I found the reason. In the last proposal, we changed the type of `kong.ctx.plugin.response_body` from **string** to **table**.

![image](https://user-images.githubusercontent.com/13045944/96861454-b3da2500-1496-11eb-8079-cccf1351862e.png)

Therefore, it's necessary to check if `kong.ctx.plugin.response_body` is nil before using `table.concat(kong.ctx.plugin.response_body)` in pkg/log.lua. This problem has now been fixed.

![image](https://user-images.githubusercontent.com/13045944/96861926-601c0b80-1497-11eb-9f10-ef4f81263782.png)


In this time, some integration tests failed: https://github.com/Kong/kong/pull/5968/checks?check_run_id=1291355165#step:6:1552, but it does not seem to be related to my changes.

Because the previous integration test failed, many later plugin tests were canceled. But from here we can see that the `01-tcp-log` test passed: https://github.com/Kong/kong/pull/5968/checks?check_run_id=1291355317#step:6:1352

It seems that there is no problem, but maybe we can rerun all the tests to make sure there are no other problems.